moped-editor/src/views/projects/projectView/ProjectComponents/utils/makeFeatureCollections.js

```diff
@@ -23,9 +23,10 @@ export const makeFeatureFromProjectGeographyRecord = (record) => ({
 export const getAllComponentFeatures = (component) => {
   const allComponentFeatures = [];
 
-  Object.keys(featureTableFieldMap).forEach((key) => {
-    if (component.hasOwnProperty(key)) {
-      allComponentFeatures.push(component[key]);
+  Object.keys(featureTableFieldMap).forEach((featureTableName) => {
+    if (component.hasOwnProperty(featureTableName)) {
+      const feature = cloneDeep(component[featureTableName]);
+      allComponentFeatures.push(feature);
```

Comment:
apparently this only becomes a problem when you enable caching—you cannot directly modify any object stored in the Apollo cache. they are immutable.

Reply:
derp—thanks for flagging this. i will update the import :/